tests/test_auth.py

```diff
@@ -0,0 +1,81 @@
+import json
+import unittest
+from flask import Flask
+
+from app import create_app, db
+from app.models import User
+
+class BaseTest(unittest.TestCase):
+    """
+    Base test to be used by each test method.
+    It enables the DRY principles to be followed.
+    """
+
+    def setUp(self):
+        self.app = create_app('testing')
+        self.client = self.app.test_client()
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+
+        # reusable instance
+        self.user = User(username='damidee', password='yello')
+
+        # create table
+        db.create_all()
+
+
+    def tearDown(self):
+        db.session.remove()
+        db.drop_all()
+        self.app_context.pop()
+
+
+class TestAuthorization(BaseTest):
+
+    def test_register_succesfully(self):
```

Comment:
You can improve the structure of each test case by following the [Arrange Act Assert](http://wiki.c2.com/?ArrangeActAssert) pattern. It looks something like this. (without the comments :))

```python
def test_register_succesfully(self):
    # Arrange
    endpoint = '/api/v1/auth/register'
    credentials = {'username': 'dee', 'password': 'hello'}
    
    # Act 
    response = self.client.post(endpoint, data=credentials)
    welcome_message = json.loads(response.get_data(as_text=True)).get('message')

    # Assert
    self.assertEqual(response.status_code, 201)
    self.assertEqual("Welcome dee to the Bucketlist Service.", welcome_message)
```

You might have noticed that I didn't include the last part 
```python
    user = User.query.filter_by(username='dee').first()
    self.assertIsNotNone(user)
    self.assertIn('dee', user.username)
```

This is because, while the first assertion checks the response from the request, this second `action and assertion` checks that the user was successfully created in the database.

That is a different concept (api response vs db contents) and should be moved to another test case.

A test should only have one reason to fail. 
REF: [One of many](http://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test)

Try as much as you can to have only one assert statement per testcase. 
Sometimes it's okay to have more than one when you are testing the same concept. e.g contents of the response from an api call
Example
```python
response = request('some_url')
self.assertEqual(response.status, '201')
self.assertEqual(response.text, 'Hello world')
```